client/client.go

```diff
@@ -568,30 +606,39 @@ func (c *Client) GetClientAlloc(allocID string) (*structs.Allocation, error) {
 
 // GetServers returns the list of nomad servers this client is aware of.
 func (c *Client) GetServers() []string {
-	endpoints := c.servers.all()
+	endpoints := c.servers.GetServers()
 	res := make([]string, len(endpoints))
 	for i := range endpoints {
-		res[i] = endpoints[i].addr.String()
+		res[i] = endpoints[i].String()
 	}
+	sort.Strings(res)
 	return res
 }
 
 // SetServers sets a new list of nomad servers to connect to. As long as one
 // server is resolvable no error is returned.
-func (c *Client) SetServers(servers []string) error {
-	endpoints := make([]*endpoint, 0, len(servers))
+func (c *Client) SetServers(in []string) error {
+	endpoints := make([]*servers.Server, 0, len(in))
 	var merr multierror.Error
-	for _, s := range servers {
+	for _, s := range in {
 		addr, err := resolveServer(s)
 		if err != nil {
 			c.logger.Printf("[DEBUG] client: ignoring server %s due to resolution error: %v", s, err)
 			merr.Errors = append(merr.Errors, err)
 			continue
 		}
 
-		// Valid endpoint, append it without a priority as this API
-		// doesn't support different priorities for different servers
-		endpoints = append(endpoints, &endpoint{name: s, addr: addr})
+		// Try to ping to check if it is a real server
+		ok, err := c.Ping(addr)
+		if err != nil {
+			merr.Errors = append(merr.Errors, fmt.Errorf("Server at address %s failed ping: %v", addr, err))
+			continue
+		} else if !ok {
```

Comment:
This will never be hit as ok is only false if err is non-nil.

I'd prefer only returning an error from Ping since a nil error implies `ok=true` and a non-file error `ok=false`

Reply:
Done